### ISSUE TITLE
violation of GFM example #287

### ISSUE THREAD

**Issue by Inilien** [markedjs/marked#1591]
**Describe the bug**
I couldn't find setings in the demo of marked which would give the expected by GFM specification behavior for the [example #287](https://github.github.com/gfm/#example-287)

**To Reproduce**
It is best to just use the link above.  

I duplicate it here though too:

```
- foo
  - bar
    - baz


      bim
```
should look like this:
- foo
  - bar
    - baz

      bim

While marked behaves like [this](https://marked.js.org/demo/?text=-%20foo%0A%20%20-%20bar%0A%20%20%20%20-%20baz%0A%0A%0A%20%20%20%20%20%20bim&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20true%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=master)

**Expected behavior**
`bim` should be just another paragraph in the last list item and not a code block

**Comment by styfle** [comment]
Thanks for reporting.

This is a known issue as seen in the test suite here (notice the shouldFail flag): https://github.com/markedjs/marked/blob/a42413d4fa29834acea871273f8076cc2bf990b4/test/specs/gfm/commonmark.0.29.json#L2238

Would you like to submit a PR to fix it?

**Comment by UziTech** [comment]
similar to #1503

**Comment by alexanderankin** [comment]
![image](https://user-images.githubusercontent.com/8921892/82752394-c20f0280-9d8b-11ea-8bda-756534fdb9c0.png)

This is the regular expression, as i understand it, that is not giving enough text back.

https://github.com/markedjs/marked/blob/7b3036f8c0440cfd003ce47dd6e1a92af0f5e822/src/rules.js#L17

i dont know what exactly needs to happen in order for it to do that, augment it with the paragraph recognizer text? Here is the actual hyper text markup it is outputting:

```html
<ul><li>foo<ul><li>bar<ul><li>baz</li></ul></li></ul></li></ul>
<pre><code>  bim</code></pre>
```
and what i managed to get it to was `<ul><li><p>foo</p></li><li><p>bar...` but it would still produce the bim in its own code tag with some spaces inside the tag.

**Comment by alexanderankin** [comment]
heres my attempt at breaking down what the regex does:

```
/
  ^           # start of string
  ( {0,3})    # first level < 4 spaces
  (bull)      # bullet point is * or -, etc. (resolved line 47)
  ' '         # space
  [\s\S]+?    # optional 1+ non/white space
  (?:         # non capturing group
    hr        # hr (also resolved later)
      |       # OR
    def       # def (also resolved later)
      |       # OR
    \n{2,}       # two+ newlines
    (?! )        # which are not followed by spaces
    (?!\1bull )  # & not flwd by same indent+bullet+space
    \n*          # are followed by any number of newlines
      |          # OR
    \s*$      # whitespace til new line
  )
/
```

maybe `whitespace til new line` needs to allow for new lines? or maybe instead add a fifth branch to the conditional. It would appear that having the other ones produce paragraphs would not satisfy the test (and may break others). anyways at this point put enough time to realize i should walk away for today.

**Comment by UziTech** [comment]
> [\s\S]+?    # optional 1+ non/white space

`+?` actually means non-greedy 1+. It will try to match the least possible instead of the most possible.

**Comment by calculuschild** [comment]
This will be fixed by #2112
